internal/discovery/service.go

```diff
@@ -0,0 +1,140 @@
+package discovery
+
+import (
+	"context"
+	"fmt"
+	"github.com/flipkart-incubator/dkv/internal/hlc"
+	"github.com/flipkart-incubator/dkv/internal/master"
+	"github.com/flipkart-incubator/dkv/internal/storage"
+	"github.com/flipkart-incubator/dkv/pkg/ctl"
+	"github.com/flipkart-incubator/dkv/pkg/serverpb"
+	"gopkg.in/ini.v1"
+	"io"
+	"strconv"
+
+	"encoding/json"
+	"go.uber.org/zap"
+)
+
+/*
+This class contains the behaviour of receiving status updates from nodes in the cluster
+and providing the latest cluster info of active master / followers / slave of a region when requested
+*/
+
+type DiscoveryConfig struct {
+	// time in seconds after which the status entry of the region & node combination can be purged
+	StatusTTl uint64
+	// maximum time in seconds for the last status update to be considered valid
+	// after exceeding this time the region & node combination can be marked invalid
+	HeartbeatTimeout uint64
+}
+
+func NewDiscoverConfigFromIni(sect *ini.Section) (*DiscoveryConfig, error) {
+	sectConf := sect.KeysHash()
+	if statusTTL, err := strconv.ParseUint(sectConf["statusTTL"], 10, 64); err == nil {
+		if heartbeatTimeout, err := strconv.ParseUint(sectConf["heartbeatTimeout"], 10, 64); err == nil {
+			return &DiscoveryConfig{statusTTL, heartbeatTimeout}, nil
+		}
+	}
```

Comment:
TODO: Figure out if HOCON can be used instead which can get remove this parsing logic